src/web/assets/cp/src/js/BaseElementIndex.js

```diff
@@ -1323,6 +1385,38 @@ Craft.BaseElementIndex = Garnish.Base.extend(
       });
     },
 
+    criteriaHasChanged: function () {
+      if (!this._previousViewParams) {
+        return false;
+      }
+
+      return !_.isEqual(
+        this._viewParams.criteria,
+        this._previousViewParams.criteria
+      );
```

Comment:
@gcamacho079 Is this the only reason for including Lodash? If so, it looks like `Craft.compare()` would work just as well for this purpose.

Reply:
@brandonkelly good call, I didn't realize that existed. I was just trying this locally, and I'm getting a couple of errors trying to implement it.

First, when I try it with these objects as parameters, I'm getting an error at some point recursively where `obj1` is null and the compare is failing. If I add a return at the beginning of the function (`if (obj1 === null || obj2 === null) return`), the `compare` returns the opposite of what I would expect. 🤔

Reply:
@gcamacho079 Doh, was able to reproduce that with `Craft.compare({foo: null}, {foo: 'foo'})`. Fixed now, including in this branch.